zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/InterpreterSetting.java

```diff
@@ -266,6 +272,14 @@ public InterpreterSetting(InterpreterSetting o) {
     this.conf = o.getConf();
   }
 
+  private void createLauncher() {
+    if (group.equals("spark")) {
+      this.launcher = new SparkInterpreterLauncher(this.conf);
```

Comment:
@zjffdu Do you think it's possible to make this part more generic? For example, read class name of launcher from interpreter-setting.json and instantiate it using reflection. So any interpreter can define their own launcher without changing zeppelin core modules?

Reply:
Yeah，I think about it before. But it may involve more work. Because I don't want to make interpreter jar mix with zeppelin server jar. Although I can use URLClassLoader, still feel a little risky to do it right now because Interpreter may not be instantiated correctly in zeppelin server process if it has some other dependencies, it requires interpreter do initialization in method open instead of its constructor.